src/components/footer/index.js

```diff
@@ -1,10 +1,30 @@
-import React from 'react';
-import githubLogo from '../../images/repo.svg';
+import React, { useState, useEffect }from 'react'
+import githubLogo from '../../images/repo.svg'
+import Progress from '../progress'
 import './Footer.css'
 
 function Footer() {
+  const [storage, setStorage] = useState({ quota: 0, usage: 0})
+
+    async function showEstimatedQuota() {
+        if (navigator.storage && navigator.storage.estimate) {
+          const estimation = await navigator.storage.estimate();
+          setStorage({quota: estimation.quota, usage: estimation.usage})
+          console.log(`Quota: ${estimation.quota}`);
+          console.log(`Usage: ${estimation.usage}`);
+        } else {
+          console.error("StorageManager not found");
```

Comment:
if you set storage to null here and on initialing on line 7 then in the render you can do `{storage && <Progress`... then it'll only display if available